src/common/GlobalBadges.cpp

```diff
@@ -0,0 +1,124 @@
+#include "GlobalBadges.hpp"
```

Comment:
I would much prefer if we were able to reuse the `TwitchBadges` class for this. This means that if the user does not have a network connection, we might not be able to provide a "pretty name" of a badge, nor an icon, but I believe that's worth it

`loadEmoteImage` can basically be replaced with `TwitchBadges->badge(badge_identifier, "")`

Reply:
`GlobalBadges` wraps TwitchBadges to manage the downloading of the actual badge image, but still uses `TwitchBadges` underneath to load https://badges.twitch.tv/v1/badges/global/display. `GlobalBadges` uses callbacks to update the UI without using a QPainter as normal badges/emotes do. I think that `GlobalBadges` is valuable because  `TwitchBadges` requires the UI to be rebuilt when emotes are loaded (currently after waiting/polling because loading an image has no signal) but `GlobalBadges` fixes those issues.

Hopefully that was coherent

Reply:
`GlobalBadges` still doesn't feel like it serves a proper purpose for me. Every useful addition could (and should) fit into the `TwitchBadges` class instead.

Add the `getBadgeIcon` variants to `TwitchBadges`.
Make `TwitchBadges` a singleton.
Modify the places that previously owned their own `TwitchBadges` instance use the singleton instead.

Reply:
:+1: